server/src/main/java/org/elasticsearch/index/shard/IndexEventListener.java

```diff
@@ -86,6 +86,26 @@ default void indexShardStateChanged(
         @Nullable String reason
     ) {}
 
+    /**
+     * Invoked before a shard performs a mutable operation. Mutable operations include, but are not limited to:
+     * <ul>
+     *     <li>Indexing operations</li>
+     *     <li>Document updates</li>
```

Comment:
nit This depends on whether @tlrx finds a way to make the update-related search work for hollow shards. If that can happen, the update later makes bulk request which fall into the "indexing operations" category.

Reply:
For now I've removed the reference to updates in https://github.com/elastic/elasticsearch/pull/122241/commits/5b6aa8731f82ae61347ac2d96ea551991688c5c9